lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -44,21 +44,36 @@ def object
         def initialize(serializer, options = {})
           super
           @include_tree = IncludeTree.from_include_args(options[:include])
-          @fieldset = options[:fieldset] || ActiveModel::Serializer::Fieldset.new(options.delete(:fields))
+          @fieldset = options[:fieldset] || ActiveModel::Serializer::Fieldset.new(options[:fields])
```

Comment:
Then I'm up to remove it in a subsequent PR, as keeping it around would force us to keep the `FieldSet` interface constant, which we may not necessarily want.